### ISSUE TITLE
TELEWORK: Send headquarter location data to data warehouse

### ISSUE THREAD

**Issue by Stella-Archer** [bcgov/digital-journeys#1930]
**User Story**
“As a PSA executive I want to know which HQ location a teleworker is associated with so I can get some idea on location use.” 

**Tasks**
In the 'Headquarters Location' section of the Telework agreement, set up the following fields:

1.  Address (WorkAddress)
2.  City (WorkCity)
3.  Postal Code (WorkPostal)
to capture data and send to the dataware house for consumption by analysts
<img width="1560" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBekVPQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--673f60db65b624693cb470c49ce498c59900d994/image.png" alt="image.png" />

**Acceptance Criteria**
The checklist the product owner uses to decide whether a story is done. Objective and discrete answers (such as a yes/no) that can be tested.
- [ ] End-to-end functional testing 
- [ ] End-to-end UX testing
- [ ] If change is data impacting, test to confirm data has been sent with BI/DIO team

**Comment by mikeshitelus** [comment]
A task on teams has been created, DIO and TES will do assessment.

**Comment by Stella-Archer** [comment]
Thanks @mikeshitelus  could you let our team know a ballpark on how much time/effort this will take?

**Comment by mikeshitelus** [comment]
@Stella-Archer There were already conversations between DIO team and TES BI, based on the email threats, Justin had the estimate on the ask (I will be on vacation starting this Friday until New Year, not sure how urgent this is).
<<
It will be “medium” effort for the TES BI team:
•	Alter the table in Datamart_Telework
•	Alter ODS
•	Alter the API endpoint
•	Manually import and process historical data (if provided)
•	Test/UAT/Deploy and etc

Thanks,
Justin
>>

**Comment by Stella-Archer** [comment]
Thanks @mikeshitelus Could you give us an actual / hours estimate prior to starting work?
thx

**Comment by mikeshitelus** [comment]
@Stella-Archer can you please assign this to Patrick as well, he is discussing the priority and effort with Mira.

**Comment by Stella-Archer** [comment]
thanks @mikeshitelus @T854236 is now assigned to tix :)

**Comment by Stella-Archer** [comment]
Hello @T854236 do you have any ballpark on time (in hours) this medium effort will take the TES team?

**Comment by notjustinchoy** [comment]
Hi, I have added the 3 fields to the existing endpoint in Test.

1. workaddress1
2. workcity
3. workpostal 

**Sizing**
workaddress1 (500)
workcity (255)
workpostal (255)

_Endpoint Info_:
https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rst#Datamart_Telework_app_telework_info

**Endpoint**:
https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rsc/Datamart_Telework_app_telework_info

If it passes testing, I will promote to Prod (analytics-api.psa.gov.bc.ca/apiserver).

Thx

**Comment by Stella-Archer** [comment]
the end points have promoted the changes to Production.

Endpoint Info:
https://analytics-api.psa.gov.bc.ca/apiserver/api.rst#Datamart_Telework_app_telework_info

Endpoint:
https://analytics-api.psa.gov.bc.ca/apiserver/api.rsc/Datamart_Telework_app_telework_info

**Comment by bhumin-fw** [comment]
@notjustinchoy @T854236 
We are using https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rsc/Datamart_Telework_employee_demo API endpoint to pull employee information for telework form.
cc: @mikeshitelus

**Comment by bhumin-fw** [comment]
@notjustinchoy @T854236 Hi, I just wanted to check in and see if there are any updates on this?

**Comment by T854236** [comment]
@bhumin-fw 

Hi Bhumin, 
the fields are available already in https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rsc/Datamart_Telework_employee_demo called office_address1, office_address2, office_city, office_postal and office_country.

**Comment by bhumin-fw** [comment]
@T854236 Yes, we pull those details. Let me use value from those fields and populate on workaddress1, workcity and workpostal fields on telework form.

**Comment by bhumin-fw** [comment]
@Stella-Archer @ayushdamani It is ready to test. It will populate employee's work location on telework form.

@T854236 @notjustinchoy I have submitted one application with these data (workaddress1, workcity, workpostal), it's application_id is 8095. could you please verify that on test server.

**Comment by Stella-Archer** [comment]
The pre-fill HQ field in theory would be great but I'm not getting the correct postal code.
my office at 810 blanshard pulls in "V8W 3E1"
but when I look it up on postal code look up it says is should be "V8W 2H2"
where is the value "V8W 3E1" coming from?
<img width="1455" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVdxQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--da8003bd58e144980ff04666f05465647837db05/image.png" alt="image.png" />

<img width="1519" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVNxQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--6f442991083003342b92e2e5fa9d4e0620c8e4d1/image.png" alt="image.png" />

**Comment by Stella-Archer** [comment]
I don't think "V8W 3E1" is correct for PO box either :(
<img width="841" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWlxQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--036ff40e341cd38b72c61405936478846c20f8f0/image.png" alt="image.png" />

**Comment by bhumin-fw** [comment]
@Stella-Archer The data is coming from the government database or the ODS database itself, where your employee details have been stored.

here @T854236 can confirm from which source this data are coming.
url: https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rsc/Datamart_Telework_employee_demo 
fields: office_address1, office_address2, office_city, office_postal and office_country.

It is currently in testing. If it looks good, we will move it to PROD with pre-population. If it doesn’t work, we will only update the field names and will not pull data from the ODS endpoint for the office location.

**Comment by Stella-Archer** [comment]
Thanks @bhumin-fw yeah I think we have to do a bit more testing to see if this is just and anomaly, and maybe check in with the data people to see how reliable this is.
@ayushdamani could you check in to the telework on test and see what postal code value it gives you for your Kamloops office and see if it is correct?

**Comment by T854236** [comment]
Hi, the office address information is from job data in PeopleSoft associated to the employee.

@bhumin-fw please correct the field information when posting, there is a typo.  Please correct workaddress to workaddress1 in the API call.

**Comment by ayushdamani** [comment]
Here is a screenshot of the address that appears for me. The postal code appears correct for the Kamloops office:
<img width="1169" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOEN0Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--69ab55bf1654b5d335f5bf80fb77af8232a6f80d/image.png" alt="image.png" />

**Comment by Stella-Archer** [comment]
@T854236 And @bhumin-fw since theOCI migration are employee information fields from Active Directory /Entra are now captured in the datawarehouse?

**Comment by bhumin-fw** [comment]
@T854236 
"@T854236 @notjustinchoy I have submitted one application with these data (workaddress1, workcity, workpostal), it's application_id = 8095. could you please verify that on test server."

I believe 'workaddress1' is the field, and there doesn't seem to be any typo for the test application. Could you please check the test data for application_id=8095? If you need more applications, I can submit a few more.

**Comment by Stella-Archer** [comment]
@mikeshitelus  @notjustinchoy @T854236  since the OCI migration are employee information fields from Active Directory /Entra now captured in the datawarehouse? 

Could we use these values? I understand they are kept very current and already have our new address on superior street populated.

**Comment by notjustinchoy** [comment]
> [@T854236](https://github.com/T854236) "[@T854236](https://github.com/T854236) [@notjustinchoy](https://github.com/notjustinchoy) I have submitted one application with these data (workaddress1, workcity, workpostal), it's application_id = 8095. could you please verify that on test server."
> 
> I believe 'workaddress1' is the field, and there doesn't seem to be any typo for the test application. Could you please check the test data for application_id=8095? If you need more applications, I can submit a few more.

application=8095 looks good

**Comment by T854236** [comment]
@Stella-Archer unfortunately not available today.  Are we sure Entra is more up to date than PeopleSoft?

We would need to create a new endpoint specific to digital journey so we don't impact other services using the demographic information.  Some changes in design.

Would it be easier to Digital Journey to pull direct from Entra?  Does it use that as part of SSO process?

**Comment by Stella-Archer** [comment]
Yes @T854236 Entra data is very current and accurate, it's showing 563 Superior street already, correct postal and business #
<img width="1023" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM2JoQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4e2275a74dd9aee1b3c7209962ca107716592ebd/image.png" alt="image.png" />

**Comment by T854236** [comment]
@Stella-Archer ok thanks.  I also reached out on sync process with PS and Entra.  PS sends data to Entra but not the other way around.

Can you confirm Digital Journey uses Entra as part of SSO and can leverage that process to get on demand information as users sign in?

**Comment by bhumin-fw** [comment]
@T854236 No, we are not using Entra as part of any communication in the digital journey.

**Comment by T854236** [comment]
@bhumin-fw @Stella-Archer Thank you and noted.  We will create a card in the BI teams board for assessment and discuss with DIO on planning.

Is there a date we are looking to aim for changing the office address info?

**Comment by Stella-Archer** [comment]
Thanks @T854236 we have a related ticket #1937 with some text/content changes going live today.

For now we will just get the content live then and remove any pre filled data, so it will be the same as the current live form. This would mean we'd just get user reported location info for now, I'm not clear on the urgency to start collecting the location data, will check back in with my leadership.

@bhumin-fw could you remove the prefilled data for now?
